### DISCUSSION TITLE
Actual Power consumption in W

### DISCUSSION THREAD

**Discussion by domsn88** [signalkraft/mypyllant-component#173] [Q&A]
I would be interested to see the actual power consumption from my arotherm+ and unitower in W as I can read in the myVAillant APP. Is this planned or even already possible with this integration ?
BR
domsn

**Comment by domsn88** [comment]
I wanted to share that I found out the following setting was missing: Fetch real-time power usage
After enabling it and restarting HA I do see the actual consumption in Watts.

I have a aroTHERM+ & Unitower & VRC720 & VR940f

**Reply by WulfgarW** [reply]
@domsn88 
Where did you add/activate "fetch real-time power usage"?
I would also like to see the actual power consumption.
By the way: Where do you see it in the myVaillant app?

**Reply by WulfgarW** [reply]
@domsn88 
I found the config page, where I activated "fetch real-time power usage" and restated HA.
But unfortunately without success. I see no power consumption information. (My heat pump is a Vaillant aroTherm Split VWL 125/5 with Unitower and VRC720.)

**Reply by xxLeoxx93** [reply]
@WulfgarW 
Did you manage it? I've the same situation. Activated but missing.

**Reply by xxLeoxx93** [reply]
> aroTHERM+ & Unitower & VRC720 & VR940f

Same combination but I don't get any consumption Data :(

**Reply by WulfgarW** [reply]
@xxLeoxx93 
Until now, I have no solution.
(I know, that the current power consumption is available via ebus (not EEBUS), but I don't see it in HA and I don't know, where to find it in the myVaillant app.)

**Reply by domsn88** [reply]
I just enabled 
"Fetch real-time statistics (not supported on every system)" and
"Fetch real-time power usage (not supported on every system)"
in the integration settings and since then I am getting the consumption data.
I have aroTHERM+ & Unitower & VRC720 & VR940f

**Reply by WulfgarW** [reply]
@domsn88 
I have the older version of the internet module VR921 (=sensonet). That's a difference between your heating system and mine.
But @xxLeoxx93 has a VR940f and get's no power consumption.
@domsn88 Can you also see the current power consumption in the myVaillant app?

**Reply by domsn88** [reply]
@WulfgarW 
yes - I see the consumption in the App and Integration. Maybe try to reload your Internet Router as there are API ratelimits from Vaillant in place which might cause this. By reloading you will get a new IP Address assigned and therefore a new API quota at Vaillant.

**Reply by WulfgarW** [reply]
@domsn88 
Can you show me a scan, where you see the power consumption in the myVaillant app?
I see the energy information for my system (if I click on the main screen in the lower right corner), but I have no idea, where the location for the current power consumption is.

**Comment by domsn88** [comment]
you are on the right track. from there click on systemview (guessing as mine is called "Systemansicht" but in german) on the top

<img width="299" alt="image" src="https://github.com/user-attachments/assets/dcf5ab90-0405-4651-b8e3-cb0519ab00f2">

and then on aroTHERM plus in the middle section like here
<img width="299" alt="image" src="https://github.com/user-attachments/assets/0ef711bc-6c5a-4bcc-8e53-46548c63236f">
<img width="299" alt="image" src="https://github.com/user-attachments/assets/4b5ad36c-11a7-4413-a3c9-be83f3c1d6d1">

**Reply by WulfgarW** [reply]
For my system, the app only shows the serial number when I open the pull down menu of aroTHERM.

**Reply by WulfgarW** [reply]
I had opend a support ticket at Vaillant, asking, if I could change something in the settings of the heat pump in order to see the data @domsn88 can see.
Today I got an answer: The software of my heat pump, produced in 2022, is (at the moment) not able to provide the data above to the myVaillant app.

**Comment by Tonio7456** [comment]
### Feature Request: Real-Time Power Consumption Sensor


Hello,

First, thank you for the amazing work on this integration—it’s been incredibly helpful!

I’d like to request the addition of a real-time power consumption sensor (in Watts) as part of the myPyllant integration. The MyVaillant app appears to display this data, and it would be incredibly useful to have this available in Home Assistant for monitoring and automation purposes.

### Proposed Implementation

1. API Data Retrieval

It seems that the API might expose this data under a key like **currentPower**. The integration could be modified to fetch this data during updates. For example, in the **coordinator.py**:

```
async def _async_update_data(self):
    data = await self.api_client.fetch_data()
    return {
        "currentPower": data.get("currentPower"),
        # Other existing keys...
    }
```

2. Sensor Definition

A new sensor entity could be added to represent the real-time power consumption. 
This might look like the following in **sensor.py**:

```
class PowerSensor(SensorEntity):
    """Sensor for real-time power consumption."""

    def __init__(self, coordinator, device_info):
        super().__init__(coordinator)
        self._attr_name = "Current Power Consumption"
        self._attr_unique_id = f"{device_info['serialNumber']}_current_power"
        self._attr_device_info = device_info
        self._attr_unit_of_measurement = "W"
    
    @property
    def state(self):
        """Return the current power consumption."""
        return self.coordinator.data.get("currentPower")
```

3. Integration in Home Assistant
Once the sensor is added, users could display it in Home Assistant dashboards or use it in automations as **sensor.current_power_consumption**.


### Benefits
Energy Monitoring: Users can track their energy consumption in real-time.
Automations: Enable automations like adjusting heating when consumption exceeds a threshold.
If this feature is already planned or in progress, I’d love to know! Otherwise, I’d be happy to assist in testing or providing further details.

Thank you for considering this feature, and let me know if additional input is required.

Best regards,

**Comment by cscholz** [comment]
I would be pleased if the current power consumption could be evaluated.

In Germany, the official electrical metering concepts include one (no. 8) in which the heat pump and the household have separate meters connected in series.
Without the actual consumption, it is not possible to evaluate/display how much of the surplus electricity from the household was actually fed in after consumption by the heat pump.

**Comment by Ahrensfd** [comment]
Its look like Vailant has remove the current power information ! Dame it :-(

The current version of the app did not show this information anymore 

Is there a way to get the information about the current power back ?

![IMG_3572](https://github.com/user-attachments/assets/1cd2f014-746d-4f49-8a3b-fe9a229549c0)

**Reply by Krusty1977** [reply]
> Its look like Vailant has remove the current power information ! Dame it :-(
> 
> The current version of the app did not show this information anymore
> 
> Is there a way to get the information about the current power back ?
> 
> ![IMG_3572](https://private-user-images.githubusercontent.com/91552404/437014402-1cd2f014-746d-4f49-8a3b-fe9a229549c0.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ2NjMzMjYsIm5iZiI6MTc1NDY2MzAyNiwicGF0aCI6Ii85MTU1MjQwNC80MzcwMTQ0MDItMWNkMmYwMTQtNzQ2ZC00ZjQ5LThhM2ItZmU5YTIyOTU0OWMwLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA4VDE0MjM0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0MTgwMTEzMDBiMzRmOTA0ZDRkMzk2MjZjYzE1MjcwN2E2MjM4MGQ2MTcwZjdhYTkyZjM4Yjc5OTcwNGI1N2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0fjY6qG_LLYbMIcAT6Vd0ff2OpiRDjcEkJX75eMQeME)

In Germany we say "solche Deppen" :-)

**Reply by Krusty1977** [reply]
In my app it looks like this...

![C45A5865-F5E4-4716-9541-D902D02F494E_1_201_a](https://github.com/user-attachments/assets/b948b65d-5fd2-4b77-889b-bfdbc2ee04ac)

But I don't have a chance to integrate it into my energy dashboard, I don't know why, it always looks like this: 

<img width="500" height="1908" alt="image" src="https://github.com/user-attachments/assets/66d3a2cd-2b9c-4933-b216-aeded5dbacc3" />

**Reply by Krusty1977** [reply]
Today a major release of the iPhone App was released with a new tab „energy“ which has a lot and good informations about the energy consumption. 

So the App is really good now and i have to take back what I said before.

But this is the Proof, that vaillant has all values to calculate the energy consumption. Hopefully this helps to male the Home Assistant Integration better …

**Reply by TommyD91** [reply]
I would be so happy if the current power in watts was finally implemented. The power is now available in the Vaillant app with the update. Additionally, it would be great if the number of cycles (on/off) could also be read out via the Homos System Integration.

**Comment by DrMabuse22** [comment]
First of all, I would also like to express my sincere thanks for the work that has gone into this integration.

I would like to revisit the topic: 
Is it foreseeable or is work being done to make the current power available as an entity? The Vaillant app displays the information and it would be great to be able to see it in Home Assistant.

**Reply by JohannesW91** [reply]
go to the integration settings, then scroll a bit down to check here:
<img width="1572" height="512" alt="image" src="https://github.com/user-attachments/assets/60f03fd0-9343-401a-939a-98add953a324" />

**Reply by DrMabuse22** [reply]
Wow, that was easy! Thanks :-)

**Reply by JohannesW91** [reply]
I was looking for this feature myself for ages :-D

**Reply by Krusty1977** [reply]
i have enabled both boxes but it still does not work... I have an Vaillant aroTHERM. In the MyVaillant App on my iPhone I see the energy consumption of heating and warm water...

<img width="500" alt="image" src="https://github.com/user-attachments/assets/20b24c95-b601-410c-965b-084e3e3313c9" />

**Reply by DrMabuse22** [reply]
I reloaded the settings and after a few minutes, the new sensors where there
<img width="305" height="489" alt="image" src="https://github.com/user-attachments/assets/8f1c56a8-9d73-4cbf-b978-2355ed4341d7" />
